tools/lsp/language.rs

```diff
@@ -655,6 +674,65 @@ pub async fn remove_binding_command(
     Ok(serde_json::to_value(()).expect("Failed to serialize ()!"))
 }
 
+pub async fn populate_command(
+    params: &[serde_json::Value],
+    ctx: &Rc<Context>,
+) -> Result<serde_json::Value> {
+    let text_document = serde_json::from_value::<lsp_types::OptionalVersionedTextDocumentIdentifier>(
+        params.first().ok_or("No text document provided")?.clone(),
+    )?;
+    let new_text =
+        serde_json::from_value::<String>(params.get(1).ok_or("No contents provided")?.clone())?;
+
+    let edit = {
+        let document_cache = &mut ctx.document_cache.borrow_mut();
+        let uri = text_document.uri;
+        let version = document_cache.document_version(&uri);
+
+        if let Some(source_version) = text_document.version {
+            if let Some(current_version) = version {
+                if current_version != source_version {
+                    return Err(
+                        "Document version mismatch. Please refresh your command data".into()
```

Comment:
This is the kind of error that is seen by the user when something goes wrong in synchronization.  But users will have no idea how to "refresh their command data".